introspection-engine/introspection-engine-tests/tests/re_introspection/relation_mode/postgres.rs

```diff
@@ -0,0 +1,376 @@
+        }
+
+        datasource db {
+          provider     = "postgres"
+          url          = env("TEST_DATABASE_URL")
+          relationMode = "prisma"
+        }
+
+        model Foo {
+          id     Int @id
+          bar    Bar @relation(fields: [bar_id], references: [id])
+          bar_id Int @unique
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
+
+    let result = api.re_introspect_config(input).await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+// referentialIntegrity="foreignKeys" is renamed as relationMode="foreignKeys", and @relations are preserved but moved to the bottom.
+#[test_connector(tags(Postgres))]
+async fn referential_integrity_foreign_keys(api: &TestApi) -> TestResult {
+    let init = formatdoc! {r#"
+        CREATE TABLE "Foo" (
+            "id" INTEGER NOT NULL,
+            "bar_id" INTEGER NOT NULL,
+        
+            CONSTRAINT "Foo_pkey" PRIMARY KEY ("id")
+        );
+        
+        CREATE TABLE "Bar" (
+            "id" INTEGER NOT NULL,
+        
+            CONSTRAINT "Bar_pkey" PRIMARY KEY ("id")
+        );
+        
+        CREATE UNIQUE INDEX "Foo_bar_id_key" ON "Foo"("bar_id");
+
+        ALTER TABLE "Foo" ADD CONSTRAINT "Foo_bar_id_fkey" FOREIGN KEY ("bar_id") REFERENCES "Bar"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+    "#};
+
+    api.raw_cmd(&init).await;
+
+    let input = indoc! {r#"
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+            provider             = "postgres"
+            url                  = env("TEST_DATABASE_URL")
+            referentialIntegrity = "foreignKeys"
+        }
+
+        model Foo {
+            id     Int @id
+            bar    Bar @relation(fields: [bar_id], references: [id])
+            bar_id Int @unique
+        }
+
+        model Bar {
+            id  Int  @id
+            foo Foo?
+        }
+    "#};
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+          provider     = "postgres"
+          url          = env("TEST_DATABASE_URL")
+          relationMode = "foreignKeys"
+        }
+
+        model Foo {
+          id     Int @id
+          bar_id Int @unique
+          bar    Bar @relation(fields: [bar_id], references: [id])
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
+
+    let result = api.re_introspect_config(input).await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+// relationMode="prisma" preserves the relation policy ("prisma") as well as @relations.
+#[test_connector(tags(Postgres))]
+async fn relation_mode_prisma(api: &TestApi) -> TestResult {
+    let init = formatdoc! {r#"
```

Comment:
nitpick: `formatdoc!()` corresponds to `format!()`, it does more work than `indoc!()`, we could use `indoc!()` in these tests (since the strings are not templates)